cmd/imagePushToRegistry.go

```diff
@@ -145,28 +146,35 @@ func copyImages(config *imagePushToRegistryOptions, utils imagePushToRegistryUti
 	g.SetLimit(10)
 	platform := config.TargetArchitecture
 
-	for i := 0; i < len(config.SourceImages); i++ {
-		src := fmt.Sprintf("%s/%s", config.SourceRegistryURL, config.SourceImages[i])
-		dst := fmt.Sprintf("%s/%s", config.TargetRegistryURL, config.TargetImages[i])
+	for _, sourceImage := range config.SourceImages {
+		sourceImage := sourceImage
+		src := fmt.Sprintf("%s/%s:%s", config.SourceRegistryURL, sourceImage, config.SourceImageTag)
 
-		g.Go(func() error {
-			log.Entry().Infof("Copying %s to %s...", src, dst)
-			if err := utils.CopyImage(ctx, src, dst, platform); err != nil {
-				return err
-			}
-			log.Entry().Infof("Copying %s to %s... Done", src, dst)
-			return nil
-		})
+		targetImage, ok := config.TargetImages[sourceImage].(string)
+		if !ok {
+			return fmt.Errorf("incorrect name of target image: %v", config.TargetImages[sourceImage])
+		}
+
+		if config.TargetImageTag != "" {
+			g.Go(func() error {
+				dst := fmt.Sprintf("%s/%s:%s", config.TargetRegistryURL, targetImage, config.TargetImageTag)
+				log.Entry().Infof("Copying %s to %s...", src, dst)
+				if err := utils.CopyImage(ctx, src, dst, platform); err != nil {
+					return err
+				}
+				log.Entry().Infof("Copying %s to %s... Done", src, dst)
+				return nil
+			})
+		}
 
 		if config.TagLatest {
 			g.Go(func() error {
-				// imageName is repository + image, e.g test.registry/testImage
-				imageName := parseDockerImageName(dst)
-				log.Entry().Infof("Copying %s to %s...", src, imageName)
-				if err := utils.CopyImage(ctx, src, imageName, platform); err != nil {
+				dst := fmt.Sprintf("%s/%s", config.TargetRegistryURL, config.TargetImages[sourceImage])
+				log.Entry().Infof("Copying %s to %s...", src, dst)
+				if err := utils.CopyImage(ctx, src, dst, platform); err != nil {
 					return err
 				}
-				log.Entry().Infof("Copying %s to %s... Done", src, imageName)
+				log.Entry().Infof("Copying %s to %s... Done", src, dst)
 				return nil
 			})
 		}
```

Comment:
I suggest to prepare `dst` var before calling `go func(...)` and to reuse the same `go func(...)`

```
var dst string
switch {
case config.TargetImageTag != "":
   dst = ...
case config.TagLatest:
   dst = ...
default: 
      return fmt.Errorf("either TargetImageTag or TagLatest should be true")
}

g.Go(func() error {
   ...
}
```

Reply:
but we can have cases when both images should be pushed (image tag is specified + tag latest is true), 
so in such cases switch does not fit well, and two goroutines are needed